module/zfs/spa.c

```diff
@@ -4106,9 +4240,36 @@ spa_create(const char *pool, nvlist_t *nvroot, nvlist_t *props,
 	    (error = vdev_create(rvd, txg, B_FALSE)) == 0 &&
 	    (error = spa_validate_aux(spa, nvroot, txg,
 	    VDEV_ALLOC_ADD)) == 0) {
-		for (int c = 0; c < rvd->vdev_children; c++) {
-			vdev_metaslab_set_size(rvd->vdev_child[c]);
-			vdev_expand(rvd->vdev_child[c], txg);
+		/*
+		 * If the ZPOOL_PROP_SPECIAL_SMALL_BLOCKS is being set, then
+		 * we need to confirm that a special vdev is present.  Avoid
+		 * dirtying a vdev if this is an invalid request.
```

Comment:
Do we make this same check when setting the property after the pool is created?

Reply:
yes